### ISSUE TITLE
misplaced-bare-raise false positive on nested try statements

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#1924]
### Steps to reproduce

Run pylint on the following code:
```python
try:
    raise ValueError('bad value')
except ValueError:
    try:
        print('cleanup')
        raise IOError('failed')
    finally:
        print('propagate')
        raise
```

This example is a bit elaborate, but I wanted to demonstrate that this is valid and useful Python code: it re-raises ValueError. The following is a minimal version of the test case:

```python
try:
    pass
except ValueError:
    try:
        pass
    finally:
        raise
```

### Current behavior

pylint reports:
```
E:  9, 8: The raise statement is not inside an except clause (misplaced-bare-raise)
```

### Expected behavior

No issue is reported by pylint.

### pylint --version output

```
No config file found, using default configuration
pylint 1.8.2, 
astroid 1.6.1
Python 2.7.13 (default, Jan 03 2017, 17:41:54) [GCC]
```

**Comment by brycepg** [comment]
Thanks for the report. I can reproduce.
